packages/playwright-crawler/src/internals/utils/playwright-utils.ts

```diff
@@ -564,7 +564,11 @@ export async function saveSnapshot(page: Page, options: SaveSnapshotOptions = {}
  */
 export async function parseWithCheerio(page: Page): Promise<CheerioRoot> {
     ow(page, ow.object.validate(validators.browserPage));
-    const pageContent = await page.content();
+
+    // eslint-disable-next-line no-new-func
+    const html = await page.evaluate(new Function(`return ${expandShadowRoots.toString()}(document)`) as any) as string;
```

Comment:
Both [Playwright](https://playwright.dev/docs/api/class-page#page-evaluate) and [Puppeteer](https://pptr.dev/guides/evaluate-javascript/) accept a string here - the libraries stringify it when sending the script to the browser anyway. 

Might save some issues with the eslint / ts here :)

Reply:
Well, for playwright I can do this and it works the same:

```ts
const html = await page.evaluate(`${expandShadowRoots.toString()}(document)`) as string;
```

With puppeteer this doesn't work.

So I guess I will rather keep the `new Function` and have the same implementation on both sides that work the same.

Reply:
Whoa, I have to keep myself from diving too deep into this, but it seems that Playwright is doing some non-standard tricks to the evaluated string...

What you're passing there is a function declaration and an expression:

`function expandShadowRoots(){} (document)` - which all of the JS runtimes I tried simply evaluate to `document`, dropping the anonymous function declaration. For some reason, Playwright doesn't(!), and evaluates this as an IIFE.

The right way of passing this would be passing just one, unambiguous IIFE expression:

`(function expandShadowRoots(){})(document)` (add parentheses around the stringified function), in which case both Playwright and Puppeteer evaluate this correctly.

Weird, but again, I refuse to look into this any deeper 👀  consistency is the key for sure.

Reply:
Right, that does the trick. I've wasted so much time dealing with this small implementation detail... :]